src/rt/lifetime.d

```diff
@@ -1193,24 +1186,53 @@ extern (C) void _d_delarray_t(void[]* p, const TypeInfo_Struct ti)
 {
     if (p)
     {
-        assert(!(*p).length || (*p).ptr);
-        if ((*p).ptr)
+        auto bic = __getBlkInfo(p.ptr);
+        auto info = bic ? *bic : GC.query(p.ptr);
+
+        if (info.base && (info.attr & BlkAttr.APPENDABLE))
```

Comment:
> Why did you add the appendable check?

Should we also free anything that was not allocated as an array by the runtime? Running dtors is dangerous in this case, though we could check flags only in case a type info is passed.